tests/conftest.py

```diff
@@ -22,12 +22,24 @@
 # THE SOFTWARE.
 #
 
+import pytest
+
 
 def pytest_addoption(parser):
     parser.addoption(
         "--adapter",
         action="store",
         default=None,
         dest="adapter",
-        help="Pass an adapter for connection to an instrument needed for a test.",
+        help="Pass an adapter string for connection to an instrument needed for a test.",
     )
+
+
+@pytest.fixture(scope="session")
+def adapter_address(pytestconfig):
```

Comment:
I'd use a name to make clear that this is for a _connected hardware device_ (hence my original suggestion "connected_HW_address"). "adapter" is used so often throughout the test suite that this fixture name will be too confusing/not clear that this is for connected devices only.

Reply:
This was indeed my doing, from back when I implemented the HMP4040 ;)

Reply:
Since, the docs state that the hardware tests should be named `_with_device`, I would say we use that here, too, and name it `_connected_device_address`.